### PR TITLE
Beta UI by pythongosssss

### LINKED ISSUE #68: [DevTask] Implement Beta UI
Sync following PRs that implements Beta UI here.

- https://github.com/comfyanonymous/ComfyUI/pull/3112
- https://github.com/comfyanonymous/ComfyUI/pull/3876
- https://github.com/comfyanonymous/ComfyUI/pull/3900
- https://github.com/comfyanonymous/ComfyUI/pull/3909

### PR DESCRIPTION
**Pull request by huchenlei**
Closes https://github.com/Comfy-Org/ComfyUI_frontend/issues/68.

@pythongosssss There is something wrong with canvas resize. See comments. Can you take a look?

### CODE REVIEW

File: src/scripts/app.ts [hunk 1]
```diff
@@ -1704,6 +1730,19 @@ export class ComfyApp {
 		await this.#invokeExtensionsAsync("setup");
 	}
 
+	resizeCanvas() {
+		// Limit minimal scale to 1, see https://github.com/comfyanonymous/ComfyUI/pull/845
+		const scale = Math.max(window.devicePixelRatio, 1);
+
+		// Clear fixed width and height while calculating rect so it uses 100% instead
+		this.canvasEl.height = this.canvasEl.width = NaN;
+		const { width, height } = this.canvasEl.getBoundingClientRect();
```
Comment (huchenlei) [inline src/scripts/app.ts:1739]: Here `width` is correctly returned by `height` is 0 for some calls.

File: src/scripts/app.ts [hunk 2]
```diff
@@ -1614,15 +1641,17 @@ export class ComfyApp {
...
-		await this.ui.settings.load();
-		await this.#loadExtensions();
 
 		// Create and mount the LiteGraph in the DOM
 		const mainCanvas = document.createElement("canvas")
 		mainCanvas.style.touchAction = "none"
 		const canvasEl = (this.canvasEl = Object.assign(mainCanvas, { id: "graph-canvas" }));
 		canvasEl.tabIndex = 1;
 		document.body.prepend(canvasEl);
+		this.resizeCanvas();
```
Comment (huchenlei) [inline src/scripts/app.ts:1651]: This call is getting correct width / height

File: src/scripts/app.ts [hunk 3]
```diff
@@ -1637,27 +1666,22 @@ export class ComfyApp {
...
-			canvasEl.getContext("2d").scale(scale, scale);
-			canvas.draw(true, true);
-		}
-
 		// Ensure the canvas fills the window
-		resizeCanvas();
-		window.addEventListener("resize", resizeCanvas);
+		this.resizeCanvas();
+		window.addEventListener("resize", () => this.resizeCanvas());
+		const ro = new ResizeObserver(() => this.resizeCanvas());
```
Comment (huchenlei) [inline src/scripts/app.ts:1680]: These calls are getting height = 0

**Review by huchenlei — COMMENTED** [review]
BTW there are also a ton of ts check errors to fix lol

**Comment by pythongosssss** [comment]
`style.css` doesnt seem to match, if I copy the file over from the main repo that fixes it

**Comment by huchenlei** [comment]
> `style.css` doesnt seem to match, if I copy the file over from the main repo that fixes it

I think the problematic css changes are from https://github.com/comfyanonymous/ComfyUI/pull/3909

**Comment by pythongosssss** [comment]
https://github.com/Comfy-Org/ComfyUI_frontend/compare/1f6eed8e736d531d5fcc32a378ce29f56125557b..fd3eb7f1eb8d449829622ea71a9aeeb97449654c

Looks like there were some additional/unintentionall changes in your merge? These seem to be the problem, merging in 3909 works fine for me

**Comment by huchenlei** [comment]
> https://github.com/Comfy-Org/ComfyUI_frontend/compare/1f6eed8e736d531d5fcc32a378ce29f56125557b..fd3eb7f1eb8d449829622ea71a9aeeb97449654c
> 
> Looks like there were some additional/unintentionall changes in your merge? These seem to be the problem, merging in 3909 works fine for me

The `@import url("scripts/ui/menu/menu.css");` is moved to .js file. Other flac check is sync from upstream.

**Pull request merged**